app/main/appium-method-handler.js

```diff
@@ -146,28 +146,38 @@ export default class AppiumMethodHandler {
     // Give the source/screenshot time to change before taking the screenshot
     await Bluebird.delay(500);
 
-    let sourceAndScreenshot;
-    if (!skipScreenshotAndSource) {
-      sourceAndScreenshot = await this._getSourceAndScreenshot();
+    let contextsSourceAndScreenshot;
+    if (!skipContextsScreenshotAndSource) {
+      contextsSourceAndScreenshot = await this._getContextsSourceAndScreenshot();
     }
 
     return {
-      ...sourceAndScreenshot,
+      ...contextsSourceAndScreenshot,
       ...cachedEl,
       res,
     };
   }
 
-  async executeElementCommand (elementId, methodName, args = [], skipScreenshotAndSource = false) {
-    return await this._execute({elementId, methodName, args, skipScreenshotAndSource});
+  async executeElementCommand (elementId, methodName, args = [], skipContextsScreenshotAndSource = false) {
+    return await this._execute({elementId, methodName, args, skipContextsScreenshotAndSource});
   }
 
-  async executeMethod (methodName, args = [], skipScreenshotAndSource = false) {
-    return await this._execute({methodName, args, skipScreenshotAndSource});
+  async executeMethod (methodName, args = [], skipContextsScreenshotAndSource = false) {
+    return await this._execute({methodName, args, skipContextsScreenshotAndSource});
   }
 
-  async _getSourceAndScreenshot () {
-    let source, sourceError, screenshot, screenshotError, windowSize, windowSizeError;
+  async _getContextsSourceAndScreenshot () {
+    let contexts, contextsError, source, sourceError, screenshot, screenshotError, windowSize, windowSizeError;
+
+    try {
+      contexts = await this.driver.contexts();
+    } catch (e) {
+      if (e.status === 6) {
```

Comment:
what is the meaning of this magic number?

Reply:
@mykola-mokhnach 

Not 100% sure about the code, never seen it so just relied on the usage of if for the other commands and copied the structure